articles/overview/what-is-qsharp.md

```diff
@@ -0,0 +1,41 @@
+
+## We want to code algorithms, not to draw circuits
+
+In [What is quantum computing?](xref:microsoft.quantum.overview.what) we saw that we can write quantum computations as quantum circuits like we do in classical computing. The circuit model has been very useful for many years in quantum computing research.
+
+However, when we write classical programs for normal computers, we don’t think in terms of circuits but in terms of logic.
+
+Here in Microsoft, we believe in empowering developers to go beyond quantum circuits by offering the tools to develop quantum algorithms. Q# allows you to use the knowledge of other high level programming languages and apply them to developing applications on quantum computers.
+
+## How does Q# work?
+
+Q# is a programming language designed to be great a writing quantum programs, by providing features that are special to quantum computing. Q# provides quantum programmers a framework that allows them to focus on the algorithms without having to care about technical details like gate sequence optimization or the physical implementation of a quantum computer.
+
+For example, one of the fundamental building blocks of Q# is the `Qubit` type, which cannot be copied or directly accessed, just like a real qubit. Instead, we can measure it and store the outcome of the measurement in a `Result` variable, a Q# type that can take two possible values: `Zero` and `One`. Constructs like this one guarantee that algorithms always respect the laws of quantum mechanics and can run correctly on quantum computers or simulators.
+
+Q# also includes classical logic features like conditionals or loops with some subtleties to make sure that all the quantum rules are being respected. For example, quantum operations need to be reversible. This enforces some constraints on the way loops are executed.
+
+Q# programs are often paired with a host program written in C# or Python, which can provide convenient organization of classical and quantum code. In addition to supporting .NET languages such as C# and Python, the QDK provides Jupyter Notebook support with the IQ# Jupyter kernel.
+
+## Use Q# to learn quantum computing
+
+Until now, to learn quantum computing you needed to learn the circuit model to understand the algorithms in the form of ordered sequences of quantum gates and measurements. With Q# you can take another path: learn quantum computing by writing quantum programs.
+
+Q# provides you with an increasing number of libraries and user-defined types that will help you to implement tools and functions already created by others to build advanced quantum algorithms. For example, you need to entangle two-qubits q1 and q2? Instead of applying individually the necessary gates to get the qubits entangled you can use the already built-in operation `PrepareEntangledState(q1, q2)`.
```

Comment:
```suggestion
Q# provides you with an increasing number of libraries and user-defined types that will help you to implement tools and functions already created by others to build advanced quantum algorithms. For example, you need to entangle two-qubits q1 and q2? Instead of applying individually the necessary gates to get the qubits entangled you can use the already built-in operation `PrepareEntangledState([q1], [q2])`.
```